extensions/Worker.Extensions.Storage.Blobs/src/BlobStorageConverter.cs

```diff
@@ -0,0 +1,272 @@
+            Type _ when targetType == typeof(IEnumerable<BlobClient>) => blobCollection.Select(b => (BlobClient)b),
+            Type _ when targetType == typeof(IEnumerable<String>) || targetType == typeof(String[]) => blobCollection.Select((b) =>  ((String)b)),
+            Type _ when targetType == typeof(IEnumerable<Stream>) => blobCollection.Select(b => (Stream)b),
+            Type _ when targetType == typeof(IEnumerable<Byte[]>) => blobCollection.Select(b => (Byte[])b),
+            Type _ when targetType == typeof(IEnumerable<BlockBlobClient>) => blobCollection.Select(b => (BlockBlobClient)b),
+            Type _ when targetType == typeof(IEnumerable<PageBlobClient>) => blobCollection.Select(b => (PageBlobClient)b),
+            Type _ when targetType == typeof(IEnumerable<AppendBlobClient>) => blobCollection.Select(b => (AppendBlobClient)b),
+            Type _ when targetType == typeof(IEnumerable<BlobContainerClient>) => blobCollection.Select(b => (BlobClient)b),
+            _ => null
+        };
+
+        private object[]? ToTargetTypeArray(Type targetType, IEnumerable<object> blobCollection) => targetType switch
+        {
+            Type _ when targetType == typeof(BlobBaseClient[]) => blobCollection.Select((b) => ((BlobBaseClient)b)).ToArray(),
+            Type _ when targetType == typeof(BlobClient[]) => blobCollection.Select((b) => ((BlobClient)b)).ToArray(),
+            Type _ when targetType == typeof(String[]) => blobCollection.Select((b) => ((String)b)).ToArray(),
+            Type _ when targetType == typeof(Stream[]) => blobCollection.Select((b) => ((Stream)b)).ToArray(),
+            Type _ when targetType == typeof(Byte[][]) => blobCollection.Select((b) => ((Byte[])b)).ToArray(),
+            Type _ when targetType == typeof(BlockBlobClient[]) => blobCollection.Select((b) => ((BlockBlobClient)b)).ToArray(),
+            Type _ when targetType == typeof(PageBlobClient[]) => blobCollection.Select((b) => ((PageBlobClient)b)).ToArray(),
+            Type _ when targetType == typeof(AppendBlobClient[]) => blobCollection.Select((b) => ((AppendBlobClient)b)).ToArray(),
+            Type _ when targetType == typeof(BlobContainerClient) => blobCollection.Select((b) => ((BlobClient)b)).ToArray(),
+            _ => null
+        };
+
+        private async Task<object?> DeserializeToTargetObject(Type targetType, string connectionName, string containerName, string blobName)
+        {
+            var content = await GetBlobStream(connectionName, containerName, blobName);
+            return _workerOptions.Value.Serializer.Deserialize(content, targetType, CancellationToken.None);
+        }
+
+        private object? DeserializeToTargetObjectCollection(IEnumerable<object> blobCollection, Type targetType)
+        {
+            (string methodName, Type type) = targetType.IsArray ? (nameof(CloneToArray), targetType.GetElementType())
```

Comment:
We're checking `targetType.IsArray` a couple times (maybe 3?) - I wonder if there is a way to refactor some of this so that we only need to check IsArray once?